snippets/core/system-text-json/csharp/RoundtripToFileAsync.cs

```diff
@@ -30,24 +29,6 @@ public static async Task RunAsync()
             }
             // </SnippetDeserialize>
             weatherForecast.DisplayPropertyValues();
-
-            using (FileStream fs = File.Create(fileNameUtf8))
```

Comment:
I see what you were trying to get at here. If the file is UTF-8 encoded (with/without BOM), the `JsonSerializer` APIs will work just fine (that is what you are doing above - that's the default behavior/expectation).

If, however, the file/stream contained data that was encoded in something else (say UTF-16), then, the `JsonSerializer APIs will fail and throw (both when the UTF-16 BOM exists or not). 

```text
// If UTF-16 BOM exists, it is considered invalid
Unhandled exception. System.Text.Json.JsonException: '0xFF' is an invalid start of a value. Path: $ | LineNumber: 0 | BytePositionInLine: 0.

// Without a UTF-16 BOM, 0x00 is invalid JSON since the JSON APIs assume UTF-8 bytes.
Unhandled exception. System.Text.Json.JsonException: '0x00' is an invalid start of a property name. Expected a '"'. Path: $ | LineNumber: 0 | BytePositionInLine: 1.
```

For reading such files, you have to read them as strings first, and you can't use the streaming APIs for that. Similarly, for writing such files, you have to first write to a UTF-8 byte[]/string first, and then write that to the file.

**I don't know if such a code sample is necessary since I don't think it will be all that common.**